packages/wallet/src/services/ProviderTracker/TrackedHandleProvider.ts

```diff
@@ -0,0 +1,37 @@
+import { BehaviorSubject } from 'rxjs';
+import { CLEAN_FN_STATS, ProviderFnStats, ProviderTracker } from './ProviderTracker';
+import { HandleProvider } from '@cardano-sdk/core';
+
+export class HandleProviderStats {
+  readonly healthCheck$ = new BehaviorSubject<ProviderFnStats>(CLEAN_FN_STATS);
+  readonly resolveHandles$ = new BehaviorSubject<ProviderFnStats>(CLEAN_FN_STATS);
+
+  shutdown() {
+    this.healthCheck$.complete();
+    this.resolveHandles$.complete();
+  }
+
+  reset() {
+    this.healthCheck$.next(CLEAN_FN_STATS);
+    this.resolveHandles$.next(CLEAN_FN_STATS);
+  }
+}
+
+/**
+ * Wraps a HandleProvider, tracking # of calls of each function
+ */
+export class TrackedHandleProvider extends ProviderTracker implements HandleProvider {
```

Comment:
Missing tests for this. Although you could just remove it entirely and just use HandleProvider directly - tracked providers are used to determine wallet's sync state (when it settles etc.)

Reply:
It will be, but not the provider that's used in TxBuilder. Wrapped (tracked) provider should only be used for resolving wallet's state, and transaction that's being built is not part of it.

Reply:
ea7c3ca3009ec6aaf161505165af8c9a2481bbb2 should address this